File: InterestingParty.java
import java.util.HashMap;
import java.util.Map;

public class InterestingParty {
  public int bestInvitation(String[] first, String[] second) {
    Map<String, Integer> map = new HashMap<>();

    for (int i = 0; i < first.length; i++) {
      String f = first[i];
      if (map.containsKey(f)) {
        int c = map.get(f);
        map.put(f, c + 1);
      } else {
        map.put(f, 1);
      }

      String s = second[i];
      if (map.containsKey(s)) {
        int c = map.get(s);
        map.put(s, c + 1);
      } else {
        map.put(s, 1);
      }
    }

    int ans = 0;
    for (String key : map.keySet()) {
      ans = Math.max(ans, map.get(key));
    }
    return ans;
  }
}

File: interesting_party.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <functional>
using namespace std;

class InterestingParty
{
public:
  int bestInvitation(vector<string> first, vector<string> second)
  {
    map<string, int> map;
    first.insert(first.end(), second.begin(), second.end());

    int ans = 0;
    for (int i = 0; i < first.size(); i++)
    {
      string f = first.at(i);
      if (map.count(f))
      {
        map.at(f) += 1;
      }
      else
      {
        map.insert(make_pair(f, 1));
      }
      ans = max(ans, map.at(f));
    }

    return ans;
  }
};
